File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/FloatPropertySerializer.java
package ru.timeconqueror.timecore.common.capability.property.serializer;

import net.minecraft.nbt.CompoundTag;

public class FloatPropertySerializer implements IPropertySerializer<Float>{
    public static final IPropertySerializer<Float> INSTANCE = new FloatPropertySerializer();
    public static final IPropertySerializer<Float> NULLABLE = IPropertySerializer.nullable(INSTANCE);

    @Override
    public void serialize(String name, Float value, CompoundTag nbt) {
        nbt.putFloat(name, value);
    }

    @Override
    public Float deserialize(String name, CompoundTag nbt) {
        return nbt.getFloat(name);
    }
}

File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/FloatPropertySerializer.kt
package ru.timeconqueror.timecore.common.capability.property.serializer

import net.minecraft.nbt.CompoundTag


object FloatPropertySerializer : IPropertySerializer<Float> {

    override fun serialize(name: String, value: Float, nbt: CompoundTag) = nbt.putFloat(name, value)
    override fun deserialize(name: String, nbt: CompoundTag) = nbt.getFloat(name)

    object Nullable : NullPropertySerializer<Float>(this)
}